packages/@react-aria/focus/src/useHasTabbableChild.ts

```diff
@@ -0,0 +1,34 @@
+/*
+ * Copyright 2022 Adobe. All rights reserved.
+ * This file is licensed to you under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License. You may obtain a copy
+ * of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under
+ * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+ * OF ANY KIND, either express or implied. See the License for the specific language
+ * governing permissions and limitations under the License.
+ */
+
+import {getFocusableTreeWalker} from './FocusScope';
+import {RefObject, useEffect, useState} from 'react';
+
+// This was created for a special empty case of a component that can have child or
+// be empty, like Collection/Virtualizer/Table/ListView/etc. When these components
+// are empty they can have a message with a tabbable element, which is like them
+// being not empty, when it comes to focus and tab order.
+//
+// This looks at the element's children and determines if any are tabbable.
+export function useHasTabbableChild(ref: RefObject<HTMLElement>): boolean {
```

Comment:
```suggestion
export function useHasTabbableChild(ref: RefObject<HTMLElement>, isEmpty: boolean, isRenderingAnything): boolean {
```
if we pass in isEmpty, we can put that into the useEffect to limit when we run the check and to re-run when the Table changes between empty and not empty
isRenderingAnything is there to account for the fact that virtualizer has rendered some content, in this case, the empty view. I'm not married to the name of this arg, and we may want both of them in a state object we pass in as the first arg before ref